### PR TITLE
Barbican workload cli

### PR DESCRIPTION
**Pull request by priyankagupte15**
Automated OS-1994, OS-1995 and OS-1999
image

### CODE REVIEW

File: tempest/api/workloadmgr/barbican/test_barbican.py [hunk 1]
```diff
@@ -2340,3 +2340,143 @@ def test_5_barbican(self):
...
+            fip = self.get_floating_ips()
+            LOG.debug("\nAvailable floating ips are {}: \n".format(fip))
+            if len(fip) < 1:
+                raise Exception("Floating ips unavailable")
+            self.set_floating_ip(fip[0], self.vm_id)
+
+            ssh = self.SshRemoteMachineConnectionWithRSAKey(fip[0])
+            self.install_qemu(ssh)
+            self.addCustomfilesOnLinuxVM(ssh, "/opt", 3)
+            md5sums_before_full = self.calculatemmd5checksum(ssh, "/opt")
```
Comment (savitha-p) [inline tempest/api/workloadmgr/barbican/test_barbican.py]: Do you need md5sum verification for workload creation tests ? If not remove this code. Also remove code related to floating ips.

File: tempest/api/workloadmgr/barbican/test_barbican.py [hunk 2]
```diff
@@ -2340,3 +2340,143 @@ def test_5_barbican(self):
             reporting.set_test_script_status(tvaultconf.FAIL)
         finally:
             reporting.test_case_to_write()
+
+    @decorators.attr(type='workloadmgr_api')
```
Comment (savitha-p) [inline tempest/api/workloadmgr/barbican/test_barbican.py]: Change the type to workloadmgr_cli

File: tempest/api/workloadmgr/barbican/test_barbican.py [hunk 3]
```diff
@@ -2340,3 +2340,143 @@ def test_5_barbican(self):
...
+    def test_6_barbican(self):
+        try:
+            test_var = "tempest.api.workloadmgr.barbican.test_"
+            tests = [[test_var + "create_workload_with_encryption_CLI", 0],
+                     [test_var + "Create_Multiple_workloads_with_same_Secret_UUID", 0],
+                     [test_var + "create_workload_with_wrong_secretUUID", 0]]
+            reporting.add_test_script(tests[0][0])
+            self.kp = self.create_key_pair(tvaultconf.key_pair_name)
+            self.vm_id = self.create_vm(key_pair=self.kp)
+            self.volumes = []
```
Comment (savitha-p) [inline tempest/api/workloadmgr/barbican/test_barbican.py]: Remove any unused variables

File: tempest/api/workloadmgr/barbican/test_barbican.py (hunk 1 above)
Comment (priyankagupte15) [reply tempest/api/workloadmgr/barbican/test_barbican.py]: sure

File: tempest/api/workloadmgr/barbican/test_barbican.py (hunk 3 above)
Comment (priyankagupte15) [reply tempest/api/workloadmgr/barbican/test_barbican.py]: updated

File: tempest/api/workloadmgr/barbican/test_barbican.py (hunk 2 above)
Comment (priyankagupte15) [reply tempest/api/workloadmgr/barbican/test_barbican.py]: updated

File: tempest/api/workloadmgr/barbican/test_barbican.py (hunk 1 above)
Comment (priyankagupte15) [reply tempest/api/workloadmgr/barbican/test_barbican.py]: updated

File: tempest/api/workloadmgr/barbican/test_barbican.py [hunk 4]
```diff
@@ -2340,3 +2340,130 @@ def test_5_barbican(self):
...
+                time.sleep(10)
+                self.wid1 = query_data.get_workload_id_in_creation(
+                    tvaultconf.workload_name)
+                workload_available = self.wait_for_workload_tobe_available(
+                    self.wid1)
+                LOG.debug("Workload ID: " + str(self.wid1))
+
+                # Show workload details using CLI command
+                rc = cli_parser.cli_returncode(
+                command_argument_string.workload_show + self.wid1 + " --insecure")
```
Comment (savitha-p) [inline tempest/api/workloadmgr/barbican/test_barbican.py]: --insecure need not be added to the command here. It gets added to the command_argument_string.py file when fetch_resources.sh is run.

Comment (savitha-p) [inline tempest/api/workloadmgr/barbican/test_barbican.py]: --insecure need not be added to the command here. It gets added to the command_argument_string.py file when fetch_resources.sh is run.

File: tempest/api/workloadmgr/barbican/test_barbican.py [hunk 5]
```diff
@@ -2340,3 +2340,130 @@ def test_5_barbican(self):
...
+                     [test_var + "create_workload_with_wrong_secretUUID", 0]]
+            reporting.add_test_script(tests[0][0])
+            self.kp = self.create_key_pair(tvaultconf.key_pair_name)
+            self.vm_id = self.create_vm(key_pair=self.kp)
+
+            self.secret_uuid = self.create_secret()
+
+            # Create workload with CLI
+            workload_create_with_encryption = command_argument_string.workload_create_with_encryption + \
+                                              " --insecure --instance instance-id=" + str(self.vm_id) + \
```
Comment (savitha-p) [inline tempest/api/workloadmgr/barbican/test_barbican.py]: Remove --insecure flag

Comment (savitha-p) [inline tempest/api/workloadmgr/barbican/test_barbican.py]: Remove --insecure flag

File: tempest/api/workloadmgr/barbican/test_barbican.py [hunk 6]
```diff
@@ -2340,3 +2340,130 @@ def test_5_barbican(self):
...
+                if rc != 0:
+                    reporting.add_test_step("Execute workload-show command", tvaultconf.FAIL)
+                    # raise Exception("Command did not execute correctly : " + str(rc))
+                    LOG.debug("Command not executed correctly : " + str(rc))
+                else:
+                    reporting.add_test_step("Execute workload-show command", tvaultconf.PASS)
+                    LOG.debug("Command executed correctly : " + str(rc))
+
+                out = cli_parser.cli_output(
+                    command_argument_string.workload_show + self.wid1 + " --insecure")
```
Comment (savitha-p) [inline tempest/api/workloadmgr/barbican/test_barbican.py]: Remove --insecure flags from all commands

Comment (savitha-p) [inline tempest/api/workloadmgr/barbican/test_barbican.py]: Remove --insecure flags from all commands

File: tempest/api/workloadmgr/barbican/test_barbican.py (hunk 4 above)
Comment (priyankagupte15) [reply tempest/api/workloadmgr/barbican/test_barbican.py]: removed

File: tempest/api/workloadmgr/barbican/test_barbican.py (hunk 5 above)
Comment (priyankagupte15) [reply tempest/api/workloadmgr/barbican/test_barbican.py]: removed

Comment (priyankagupte15) [reply tempest/api/workloadmgr/barbican/test_barbican.py]: removed

File: tempest/api/workloadmgr/barbican/test_barbican.py (hunk 6 above)
Comment (priyankagupte15) [reply tempest/api/workloadmgr/barbican/test_barbican.py]: removed

Comment (priyankagupte15) [reply tempest/api/workloadmgr/barbican/test_barbican.py]: removed

File: tempest/api/workloadmgr/barbican/test_barbican.py (hunk 4 above)
Comment (priyankagupte15) [reply tempest/api/workloadmgr/barbican/test_barbican.py]: removed

**Review by savitha-p — APPROVED** [review]

**Review by nasharatm — APPROVED** [review]

**Review by SuhasTA4 — APPROVED** [review]

**Pull request merged**